slack/config-types.bal

```diff
@@ -0,0 +1,438 @@
+    # Configurations associated with request pooling
+    PoolConfiguration poolConfig?;
+    # HTTP caching related configurations
+    CacheConfig cache = {};
+    # Specifies the way of handling compression (`accept-encoding`) header
+    Compression compression = COMPRESSION_AUTO;
+    # Configurations associated with the behaviour of the Circuit Breaker
+    CircuitBreakerConfig circuitBreaker?;
+    # Configurations associated with retrying
+    RetryConfig retryConfig?;
+    # Configurations associated with inbound response size limits
+    ResponseLimitConfigs responseLimits = {};
+    #SSL/TLS-related options
+    ClientSecureSocket secureSocket?;
+    # Proxy server related options
+    ProxyConfig proxy?;
+    # Enables the inbound payload validation functionalty which provided by the constraint package. Enabled by default
+    boolean validation = true;
+|};
+
+# Represents token for Bearer token authentication.
+#
+# + token - Bearer token for authentication
+public type BearerTokenConfig record {|
+    string token;
+|};
+
+# Represents the credential-bearing methods.
+public enum CredentialBearer {
+    AUTH_HEADER_BEARER,
+    POST_BODY_BEARER
+}
+
+# Represents the configurations of the client used to call the introspection endpoint.
+#
+# + httpVersion - The HTTP version of the client
+# + customHeaders - The list of custom HTTP headers
+# + customPayload - The list of custom HTTP payload parameters
+# + auth - The client auth configurations
+# + secureSocket - SSL/TLS-related configurations
+public type ClientConfiguration record {|
+    Oauth2HttpVersion httpVersion = HTTP_1_1;
+    map<string> customHeaders?;
+    string customPayload?;
+    ClientAuth auth?;
+    SecureSocket secureSocket?;
+|};
+
+# Represents the HTTP versions.
+public enum Oauth2HttpVersion {
+    HTTP_1_1,
+    HTTP_2
+}
+
+# Represents the the authentication configuration types for the HTTP client used for token introspection.
+public type ClientAuth ClientCredentialsGrantConfig|PasswordGrantConfig|RefreshTokenGrantConfig;
+
+# Represents the data structure, which is used to configure the OAuth2 client credentials grant type.
+#
+# + tokenUrl - Token URL of the token endpoint
+# + clientId - Client ID of the client authentication
+# + clientSecret - Client secret of the client authentication
+# + scopes - Scope(s) of the access request
+# + defaultTokenExpTime - Expiration time (in seconds) of the tokens if the token endpoint response does not contain an `expires_in` field
+# + clockSkew - Clock skew (in seconds) that can be used to avoid token validation failures due to clock synchronization problems
+# + optionalParams - Map of the optional parameters used for the token endpoint
+# + credentialBearer - Bearer of the authentication credentials, which is sent to the token endpoint
+# + clientConfig - HTTP client configurations, which are used to call the token endpoint
+public type ClientCredentialsGrantConfig record {|
```

Comment:
This record can be removed since it is not used

Reply:
Done